### PR TITLE
add promise-chaining #22

### PR DESCRIPTION
**Pull request by NeboCode**

### CODE REVIEW

File: docs/promise-chaining.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Promise Chaining
+
+Imagine that we have a sequence of asynchronous tasks that we would love to be done one after another, while passing the result through the promise chain. This can be very useful for something like loading scripts or other resources. Promises provide a couple of simple ways to achieve that.
+
+Lets take the code we executed from the JS file and analyze it step by step:
+
+    new Promise(function(resolve, reject) { 
+    setTimeout(() => resolve(1), 1000);
+Promise is set to a timeout of 1000 ms, which is 1 second.
```
Comment (Swap76) [inline docs/promise-chaining.md]: Add code in such a way in .md files
```
// Code here
```

File: docs/promise-chaining.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    return res * 2;
+We have another `.then` handler, which is taking the previous handlers result, which is `2`. This code alerts `Value: 2`. Again, the result is returned as the result of the promise and being multiplied by 2, so in the next handler, the `res` argument equals `4`.
+
+    }).then(function(res) {
+    alert("Value: " + res); 
+    });
+We received `4` from the last handler and our last alert is with `Value: 4`.
+Execution ends with three alerts: `Value: 1`, `Value: 2` and `Value: 4` since we multiplied the argument by `2` at each step.
+
+## How it works
```
Comment (Swap76) [inline docs/promise-chaining.md]: @NeboCode  Good documentation overall

File: docs/promise-chaining.md (hunk 1 above)
Comment (NeboCode) [reply docs/promise-chaining.md]: Updated.

Github API has been rather limiting in the requests I can send through JavaScript console, so I used an open API for a Library. I hope that is okay.

File: JavaScript_Advance/promise-chaining.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    var request = new XMLHttpRequest(); // We define a new request.
+    request.open('GET', 'https://openlibrary.org/api/books?bibkeys=ISBN:0201558025&format=json'); // Defining the API endpoint and the request method.
+    request.send(); // Request is sent.
+    request.onload = () => request.status == 200 ? resolve(request) : reject(request); // Check if request succeeded (response 200 = OK), otherwise, reject the promise.
+    })
+    .then(function(result) {
+        return(result.response); // Due to the resolve(request), request is being passed down as a JSON object in the variable result, we then get the part that we need and pass it further (request response)
+    })
+    .then(function(result) {
+    return JSON.parse(result); // We parse the JSON response from the API and return it.
```
Comment (Swap76) [inline JavaScript_Advance/promise-chaining.js]: Hey please indent the content with tabs which is were

1 Tab = 2 Spaces

File: docs/promise-chaining.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    alert("Value: " + res);
+    return res * 2;
+Handler `.then` is called taking the promise result `res = 1` as an argument. Result of our initial promise is `1`, so this code alerts `Value: 1` and returns `2` to be passed to the next handler.
+
+    }).then(function(res) {
+    alert("Value: " + res);
+    return res * 2;
+We have another `.then` handler, which is taking the previous handlers result, which is `2`. This code alerts `Value: 2`. Again, the result is returned as the result of the promise and being multiplied by 2, so in the next handler, the `res` argument equals `4`.
+
+    }).then(function(res) {
```
Comment (Swap76) [inline docs/promise-chaining.md]:
```
}).then(function(res) {
    alert("Value: " + res);
    return res * 2;
```

In this part the }) from the .md as it it looking like without work you have made the .md

File: docs/promise-chaining.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    new Promise(function(resolve, reject) { 
+    setTimeout(() => resolve(1), 1000);
+Promise is set to a timeout of 1000 ms, which is 1 second.
+
+    }).then(function(res) {
+    alert("Value: " + res);
+    return res * 2;
+Handler `.then` is called taking the promise result `res = 1` as an argument. Result of our initial promise is `1`, so this code alerts `Value: 1` and returns `2` to be passed to the next handler.
+
+    }).then(function(res) {
```
Comment (Swap76) [inline docs/promise-chaining.md]: Also, indent the content everywhere in .js as well as .md files

**Comment by Swap76** [comment]
@NeboCode The changes are for all of us if you want to continue the work please reopen the request

**Comment by NeboCode** [comment]
No thanks. I'll pass on this, since I am busy with some other matters. 
Best of luck.

**Comment by Swap76** [comment]
@NeboCode The work you have done for the project is highly appreciated. Thanks :-)

Looking forward to working with you on my new project related to Deno https://github.com/Swap76/Learn-Deno/ and together making this better.

Star to the repo is highly appreciated.

**Pull request closed without merging**